### PR TITLE
BUGZ-322 - high_resolution_clock behaves differently on windows and linux

### PR DESCRIPTION
**Pull request by roxanneskelly**
on linux, time_since_epoch returns time since linux epoch,
on windows, it returns time since system start.
Math between those values on two different types of machines
was coming out bad.

To deal with this problem, the processing time for domain checkins is calculated on the domain server using the high_resolution_clock, as that is what is used to tag incoming packets with their timestamps.  That value is sent to the interface for logging.  Other calculations are done via the system_clock, for which time_since_epoch always returns time since UNIX epoch.

### CODE REVIEW

File: libraries/networking/src/NodeList.cpp [hunk 1]
```diff
@@ -649,21 +649,20 @@ void NodeList::processDomainServerList(QSharedPointer<ReceivedMessage> message)
...
     packetStream >> connectRequestTimestamp;
 
-    quint64 domainServerRequestReceiveTime;
-    packetStream >> domainServerRequestReceiveTime;
-
     quint64 domainServerPingSendTime;
     packetStream >> domainServerPingSendTime;
 
+    quint64 domainServerCheckinProcessingTime;
+    packetStream >> domainServerCheckinProcessingTime;
```
Comment (ZappoMan) [inline libraries/networking/src/NodeList.cpp]: Is this an intentional protocol change? It looks you just changed the order of `domainServerPingSendTime` and `domainServerCheckinProcessingTime` -- did you change it on both sides (client and server) and do we need to bump the protocol so old clients and new servers don't get confused?

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19711/HighFidelity-Beta-PR15653-4d8607a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19711/HighFidelity-Beta-Interface-PR15653-4d8607a.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19711/HighFidelity-Beta-PR15653-4d8607a.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19711/HighFidelity-Quest-Beta-PR15653-4d8607a.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19711/HighFidelity-Beta-PR15653-4d8607a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19711/HighFidelity-Beta-Interface-PR15653-4d8607a.exe

File: libraries/networking/src/NodeList.cpp (hunk 1 above)
Comment (roxanneskelly) [reply libraries/networking/src/NodeList.cpp]: This is an intentional change.  I didn't do the whole protocol-change thing as it only changed the semantic meaning of two of the values, not the overall packet structure.  Those two values are only used in logging and currently are incorrect between windows clients and linux servers, so I didn't mark this as a real protocol change as it doesn't do anything beyond fix the logging.

I can bump the protocol version if people think it's the right thing to do.

Comment (ZappoMan) [reply libraries/networking/src/NodeList.cpp]: ok - makes sense... no worries.

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19728/HighFidelity-Beta-PR15653-4d8607a.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19728/HighFidelity-Quest-Beta-PR15653-4d8607a.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19728/HighFidelity-Beta-PR15653-4d8607a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19728/HighFidelity-Beta-Interface-PR15653-4d8607a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19728/HighFidelity-Beta-PR15653-4d8607a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19728/HighFidelity-Beta-Interface-PR15653-4d8607a.exe

**Pull request merged**